### ISSUE TITLE
Waypoints: Allow to add them inbetween

### ISSUE THREAD

**Issue by scaidermern** [osmandapp/Osmand#4122]
Currently waypoints can only be added at the end (replacing the destination or after the current destination or right before the destination as last intermediate destination) or at the start (as first intermediate destination).

It would be nice if they could be added inbetween arbitrary other waypoints. Web routers usually support moving the route to a specific point, then placing a waypoint at this position. Such a moving of a route would be really nice. Alternatively allow to add a waypoint and add an option like "Add as nearest intermediate destination" (a better name, anyone?) which means going through all waypoints and selecting the two that are closest to the new waypoint, placing the new one inbetween them.

Here is an example:
![photo_2017-07-16_16-27-56](https://user-images.githubusercontent.com/1040784/28248364-c95fcb1a-6a43-11e7-9fe1-309cacf15dc3.jpg)

Currently it is not possible to add a new waypoint at the marker's position in such a way that the waypoint comes after waypoint 2 but before waypoint 3.

**Comment by vshcherb** [comment]
You can always change position (reorder) in the Waypoints dialog

**Comment by scaidermern** [comment]
Yes, reordering in the waypoints dialog works. However this is not very user-friendly ("what number does the waypoint have and between which other numbers do I have to move it again?") :)

**Comment by rindlerblabla** [comment]
@scaidermern You can also arrange the waypoints in order by clicking "Sort door-to-door", which is under the three point menu.

**Comment by scaidermern** [comment]
> You can also arrange the waypoints in order by clicking "Sort door-to-door", which is under the three point menu.

This is nice. And this means that the required sorting is already implemented, it just needs to be accessible from the waypoint add menu. Just add this new waypoint "door-to-door" to the existing waypoints.
